scipy/optimize/tests/test_cython_optimize.py

```diff
@@ -0,0 +1,90 @@
+"""
+Test Cython optimize zeros API functions: ``bisect``, ``ridder``, ``brenth``,
+and ``brentq``, in `scipy.optimize.cython_optimize`, by finding the roots of a
+3rd order polynomial given a sequence of constant terms, ``a0``, and fixed 1st,
+2nd, and 3rd order terms in ``args``.
+
+.. math::
+
+    f(x, a0, args) =  ((args[2]*x + args[1])*x + args[0])*x + a0
+
+The 3rd order polynomial function is written in Cython and called in a Python
+wrapper named after the zero function. See the private ``_zeros`` Cython module
+in `scipy.optimize.cython_optimze` for more information. 
+"""
+
+from __future__ import division, print_function, absolute_import
+import numpy.testing as npt
+from scipy.optimize.cython_optimize import _zeros
+
+# constants
+A0 = tuple(-2.0 - x/10.0 for x in range(10))  # constant term
+ARGS = (0.0, 0.0, 1.0)  # 1st, 2nd, and 3rd order terms
+XLO, XHI = 0.0, 2.0  # first and second bounds of zeros functions
+# absolute and relative tolerances and max iterations for zeros functions
+XTOL, RTOL, MITR = 0.001, 0.001, 10
+EXPECTED = [
```

Comment:
For `x**3-a`, these EXPECTED values seem a little off, differing by up to `1e-3` from the correct values.  That is within the tolerance XTOL.  But store the exact solution so that the tolerances can be updated later without having to change the EXPECTED.

```
np.linspace(2, 2.9, 10) ** (1.0/3.0) = 
    [1.2599210498948732,
     1.2805791649874942,
     1.300591446851387,
     1.3200061217959123, 
     1.338865900164339, 
     1.3572088082974532,
     1.375068867074141,
     1.3924766500838337, 
     1.4094597464129783,
     1.4260431471424087]
```